api/tbcapi/tbcapi.go

```diff
@@ -151,16 +151,16 @@ type TxByIdRawResponse struct {
 }
 
 type TxByIdRequest struct {
-	TxId api.ByteSlice `json:"tx_id"`
+	TxId api.ByteSlice `json:"tx_id"` // reverse order
 }
 
 type TxByIdResponse struct {
-	Tx    Tx              `json:"tx"`
+	Tx    *Tx             `json:"tx"`
```

Comment:
Same here.

Reply:
If we don't use pointers here, it results in an empty Tx being returned, which is confusing for clients.
With a pointer, the JSON will be `{"tx": null}` if the tx is empty, whereas without a pointer it becomes:
```json
{"tx": {"version": 0, "lock_time": 0, "tx_in": null, "tx_out": null}}
```